lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -1055,6 +1055,29 @@ void SwiftASTContext::DiagnoseWarnings(Process &process, Module &module) const {
     process.PrintWarningCantLoadSwiftModule(module, message);
 }
 
+/// Locate the swift-plugin-server for a plugin library.
+/// FIXME: move this to Host, it may be platform-specific.
+static std::string GetPluginServer(llvm::StringRef plugin_library_path) {
+  llvm::StringRef path = llvm::sys::path::parent_path(plugin_library_path);
+  if (llvm::sys::path::filename(path) != "plugins")
+    return {};
+  path = llvm::sys::path::parent_path(path);
+  if (llvm::sys::path::filename(path) != "host")
+    return {};
+  path = llvm::sys::path::parent_path(path);
+  if (llvm::sys::path::filename(path) != "swift")
+    return {};
+  path = llvm::sys::path::parent_path(path);
+  if (llvm::sys::path::filename(path) != "lib")
+    return {};
```

Comment:
Probably there are 2 `-plugin-path`,`${toolchain}/usr/lib/swift/host/plugins` and `${toolchain}/usr/local/lib/swift/host/plugins`. Both should use `${toolchain}/usr/bin/swift-plugin-server`
```
  if (llvm::sys::path::filename(path) == "local")
    path = llvm::sys::path::parent_path(path);
```
Also maybe 
```
  if (llvm::sys::path::filename(path) != "usr")
    return {};
```